girder/api/rest.py

```diff
@@ -661,15 +662,19 @@ def endpointDecorator(self, *path, **params):
         except cherrypy.HTTPRedirect:
             raise
         except Exception:
-            # These are unexpected failures; send a 500 status
-            logger.exception('500 Error')
+            # These are unexpected failures; send a 500 status and a UID to match with log
+            uid = str(uuid.uuid4())
+            logger.exception('500 Error (%s)' % uid)
             cherrypy.response.status = 500
-            t, value, tb = sys.exc_info()
-            val = {'message': '%s: %s' % (t.__name__, repr(value)),
-                   'type': 'internal'}
-            curConfig = config.getConfig()
-            if curConfig['server']['mode'] != 'production':
-                # Unless we are in production mode, send a traceback too
+            val = dict(type='internal', uid=uid)
+
+            if config.getConfig()['server']['mode'] == 'production':
+                # Sanitize errors in production mode
+                val['message'] = 'An unexpected error occurred on the server.'
```

Comment:
I suggest a more friendly message with a recommended action.

```
We are sorry, an unexpected error occurred on the server. Please, contact the administrator at xxxx and share the identifier yyyy to help them lookup related information.
```

Reply:
+1 I would recommend passing UUID so that users can share that info with admins.

Reply:
The UID is passed in a `uid` field of the response document, rather than in-band within the message. I think it's good to keep it there and have display of the UID be a client-side consideration and keep the `message` field to strictly human-readable stuff.

Reply:
> have display of the UID be a client-side consideration 

Make sense

Should the update of the client side be part of the PR ?

Reply:
I don't think I want to update the Backbone client with this PR. Most of our error reports involve people posting the full JSON response of the failing request anyway.

Since this is a UX concern I'd defer to @jtomeck about the best way to present this to users, though it is likely different for each Girder-based front end.